component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java

```
@@ -242,7 +245,27 @@ public Response getUserAvatarById(@Context UriInfo uriInfo,
     builder.cacheControl(cc);
     return builder.cacheControl(cc).build();
   }
-  
+
+  private Response getDefaultUserAvatar(String avatarUrl) {
```

The case default=404 is not handled. It should return an error 404 if the "default" param equals 404.